modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/WeightedSampling.scala

```diff
@@ -0,0 +1,43 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.MonadThrow
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Interval
+
+object WeightedSampling {
+
+  type ProbabilityRefinement = Interval.ClosedOpen[0.0, 1.0]
+  type Probability = Double Refined ProbabilityRefinement
+
+  type WeightRefinement = Interval.OpenClosed[0.0, 1.0]
+  type Weight = Double Refined WeightRefinement
+
+  def sample[F[_]: MonadThrow, A](chosenP: Probability)(distribution: List[(A, Weight)]): F[A] = {
+    type Agg = (Double, List[(A, Double)])
+
+    val sum = distribution.foldLeft(0.0)(_ + _._2)
```

Comment:
when possible, please avoid `_1` and `_2`. it's better to use `case (_, b) => ` naming `b` accordingly

Reply:
Changed to `distribution.foldLeft(0.0) { case (sum, (_, weight)) => sum + weight }`.